accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -1162,17 +1174,17 @@ evidence this will be useful.
 
 Thanks to expression statements, a switch expression could appear in the same
 position as a switch statement. This isn't technically ambiguous, but requires
-unbounded lookahead to tell if a switch in statement position is a statement or
-expression.
+unbounded lookahead to read past the value expression to the first `case` in
```

Comment:
Technically this isn't a problem anymore, because the scanner tags every `(` with a pointer to the matching `)`, so at parse time we *could* check for the presence of `case` or `default` in a bounded amount of time.

It might still be reasonable to prohibit switch expressions at statement level just to avoid user confusion, though.

Reply:
Our scanner might work fine, but there are other Dart scanners and parsers out there for things like syntax highlighters that may find it challenging to handle correctly.